src/ops/inventory/ec2inventory.py

```diff
@@ -39,6 +37,25 @@ def __init__(self, boto_profile, regions, filters={}, bastion_filters={}):
         # Index of hostname (address) to instance ID
         self.index = {}
 
+    def create_boto3_session(self, profile_name):
+        try:
+            # Use the profile to create a session
+            session = boto3.Session(profile_name=profile_name)
+
+            # Verify region
+            if not self.regions:
+                self.regions = [session.region_name]
+                if not self.regions[0]:
```

Comment:
I would switch the places:
`if not self.regions:
    if not session.region_name:
        raise NoRegionError
    self.regions = [session.region_name]`